java/fury-core/src/main/java/io/fury/serializer/Serializers.java

```diff
@@ -429,38 +493,82 @@ public StringBuilder read(MemoryBuffer buffer) {
     }
   }
 
-  public static final class StringBufferSerializer extends Serializer<StringBuffer> {
-    private final StringSerializer stringSerializer;
+  static Tuple2<ToByteFunction, Function> bufferCache;
+
+  private static synchronized Tuple2<ToByteFunction, Function> getBufferFunc() {
+    if (bufferCache == null) {
+      Function getValue =
+          (Function) Functions.makeGetterFunction(StringBuffer.class.getSuperclass(), "getValue");
+      if (Platform.JAVA_VERSION > 8) {
+        ToByteFunction getCoder =
+            (ToByteFunction)
+                Functions.makeGetterFunction(StringBuffer.class.getSuperclass(), "getCoder");
+        bufferCache = Tuple2.of(getCoder, getValue);
+      } else {
+        bufferCache = Tuple2.of(null, getValue);
+      }
+    }
+    return bufferCache;
+  }
+
+  public abstract static class AbstractStringBufferSerializer<T> extends Serializer<T> {
```

Comment:
This seems unnecessary, `StringBufferSerializer` can just extends `AbstractStringBuilderSerializer`. Both `StringBuffer` and `StringBuilder` share same parent class

Reply:
thanks,it should be